R/sql-generic.R

```diff
@@ -51,12 +51,13 @@ sql_join.DBIConnection <- function(con, x, y, vars, type = "inner", by = NULL, .
     inner = sql("INNER JOIN"),
     right = sql("RIGHT JOIN"),
     full = sql("FULL JOIN"),
+    cross = sql("CROSS JOIN"),
     stop("Unknown join type:", type, call. = FALSE)
   )
 
   select <- sql_join_vars(con, vars)
 
-  on <- sql_vector(
+  on <- if (length(by$x) + length(by$y) <= 0) NULL else sql_vector(
```

Comment:
Can you please pull this out into:

```R
if (...) {
  on <- ...
} else {
  on <- ...
}
```

Or maybe even consider pulling out into own function?

Reply:
Changed.